flixel/input/actions/FlxSteamController.hx

```diff
@@ -0,0 +1,242 @@
+class FlxSteamController
+{
+	private static var controllers:Array<FlxSteamControllerMetadata>;
+	
+	private static function clear()
+	{
+		#if steamwrap
+		for (i in 0...controllers.length)
+		{
+			controllers[i].active = false;
+		}
+		#end
+	}
+	
+	private static function init()
+	{
+		controllers = [];
+		#if steamwrap
+		
+		for (i in 0...Steam.controllers.MAX_CONTROLLERS)
+		{
+			controllers.push(new FlxSteamControllerMetadata());
+		}
+		
+		//If the FlxSteamUpdater hasn't been added to FlxG.inputs yet, 
+		//we need to do so now to ensure that steam controllers will
+		//be properly updated every frame
+		var steamExists = false;
+		for (input in FlxG.inputs.list)
+		{
+			if (Std.is(input, FlxSteamUpdater))
+			{
+				steamExists = true;
+				break;
+			}
+		}
+		if (!steamExists)
+		{
+			FlxG.inputs.add(new FlxSteamUpdater());
+		}
+		
+		#end
+	}
+	
+	private static function getActionSetHandle(name:String):Int
+	{
+		#if steamwrap
+		return Steam.controllers.getActionSetHandle(name);
+		#end
+		return -1;
+	}
+	
+	private static function getCurrentActionSet(SteamControllerHandle:Int):Int
+	{
+		#if steamwrap
+		if (SteamControllerHandle >= 0 && SteamControllerHandle <= controllers.length)
+		{
+			return controllers[SteamControllerHandle].actionSet;
+		}
+		#end
+		return -1;
+	}
+	
+	private static function activateActionSet(SteamControllerHandle:Int, ActionSetHandle:Int)
+	{
+		#if steamwrap
+		if (SteamControllerHandle == FlxInputDeviceID.NONE) return;
+		if (SteamControllerHandle == FlxInputDeviceID.ALL)
+		{
+			for (i in 0...controllers.length)
+			{
+				controllers[SteamControllerHandle].actionSet = ActionSetHandle;
+			}
+		}
+		else if (SteamControllerHandle == FlxInputDeviceID.FIRST_ACTIVE)
+		{
+			controllers[SteamControllerHandle].actionSet = ActionSetHandle;
+		}
+		Steam.controllers.activateActionSet(SteamControllerHandle, ActionSetHandle);
+		#end
+	}
+	
+	
+	private static function getFirstActiveHandle():Int
+	{
+		#if steamwrap
+		for (i in 0...controllers.length)
+		{
+			if (controllers[i].active)
+			{
+				return controllers[i].handle;
+			}
+		}
+		#end
+		return -1;
+	}
+	
+	private static function getConnectedControllers():Array<Int>
+	{
+		#if steamwrap
+		var arr = Steam.controllers.getConnectedControllers();
+		for (i in 0...Steam.controllers.MAX_CONTROLLERS)
+		{
+			if (arr.length > i)
+			{
+				controllers[i].handle = arr[i];
+			}
+		}
+		return arr;
+		#else
+		return [];
+		#end
+	}
+	
+	#if steamwrap
+	private static function getAnalogActionData(controller:Int, action:Int, ?data:ControllerAnalogActionData):ControllerAnalogActionData
+	{
+		data = Steam.controllers.getAnalogActionData(controller, action, data);
+		if (controller >= 0 && controller < controllers.length)
+		{
+			if (data.bActive > 0 && data.x != 0 || data.y != 0)
+			{
+				controllers[controller].active = true;
+			}
+		}
+		return data;
+	}
+	#else
+	private static function getAnalogActionData(controller:Int, action:Int):Dynamic
```

Comment:
maybe add here third optional argument `?data:Dynamic` for better match with steamwrap version of the method?